cpp/include/cudf/table/row_operators.cuh

```diff
@@ -401,6 +401,32 @@ class element_hasher {
   }
 };
 
+template <template <typename> class hash_function, bool has_nulls = true>
+class element_hasher_with_seed {
+ public:
+  __device__ element_hasher_with_seed()
+    : _seed{0}, _null_hash(std::numeric_limits<hash_value_type>::max())
+  {
+  }
+  __device__ element_hasher_with_seed(
+    uint32_t seed = 0, hash_value_type null_hash = std::numeric_limits<hash_value_type>::max())
+    : _seed{seed}, _null_hash(null_hash)
+  {
+  }
+  // seed, null_hash, byte endianness
+  template <typename T>
+  __device__ inline hash_value_type operator()(column_device_view col, size_type row_index)
+  {
+    if (has_nulls && col.is_null(row_index)) { return _null_hash; }
+
+    return hash_function<T>{_seed}(col.element<T>(row_index));
+  }
```

Comment:
Just as a heads up on handling nested types here.  One way to add support without requiring recursive examination here would be:

- Preprocess any nested type column into a uint32_t column of pre-hashed values.  

- Substitute that preprocessed column into the table view in place of the nested type. You'd also have to know not to invoke hash_function<> here and just return the value directly.

- Depending what it means to hash something like a  List<List<Struct<int, float>, int, List<float>>>>> etc you could then potentially generate this preprocessed column using the standard nested type technique of processing each level of nesting as a separate chunk of GPU work, and then recursing on the CPU.  A good example of this being something like:
https://github.com/rapidsai/cudf/blob/71d4c34e93048f9dcb9c7a53ea5a0a12931d3c37/cpp/src/structs/copying/concatenate.cu#L40
The plausibility great depends on what it even means to hash data like this, of course.